# Operator in Python
> In Python programming, Operators in general are used to perform operations on values and variables.

* > Types of Operators in Python

1. Arithmatic Operator
2. Assignment Operator
3. Comparision Operator
4. Logical Operator
5. Identity Operator
6. Membership Operator
7. Bitwise Operator


* Arithmetic Operator :- ( + , - , * , / , % , ** , // )
> Python Arithmetic operators are used to perform basic mathematical operations like addition, subtraction, multiplication, and division.

In [34]:
a=20
b=10
print(a+b)  # Addition: adds two operands

30


In [35]:
a=20
b=15
print(a-b)  # Subtraction: subtracts two operands	

5


In [36]:
a=30
b=5
print(a*b)  # Multiplication: multiplies two operands	

150


In [37]:
a=30
b=5
print("The value with decimal by default : ",a/b) 

The value with decimal by default :  6.0


In [38]:
a=30
b=13
print(a%b)  # Modulus: returns the remainder when the first operand is divided by the second	

4


In [39]:
a=5
b=3
print(a**b)  # Power: Returns first raised to power second	

125


In [40]:
a=30
b=4
print(a//b)  # Division (floor): divides the first operand by the second But remove decimal value.	

7


* Assignment Operator :- ( = , += , -= , *= , /= , %= , **= , //= )
> Python Assignment operators are used to assign values to the variables.

In [41]:
a=10
print(a)

10


In [42]:
a=a+1
print(a)

11


In [43]:
a+=1 
print(a)

12


In [44]:
a+=2
print(a)

14


In [45]:
a=a-2
print(a)

12


In [46]:
a-=2
print(a)

10


In [47]:
a=a*5
print(a)

50


In [48]:
a*=2
print(a)

100


In [49]:
a-=50
print(a)

50


In [50]:
a=a/2
print(a)

25.0


In [51]:
a/=5
print(a)

5.0


In [52]:
a*=10
print(a)

50.0


In [53]:
a=a%6
print(a)

2.0


In [54]:
a+=20
print(a)

22.0


In [55]:
a%=13
print(a)

9.0


In [56]:
a=a**2
print(a)

81.0


In [57]:
a=a//5
print(a)

16.0


In [58]:
a//=3
print(a)

5.0


In [62]:
a=400
a/=13
print(a)

30.76923076923077


In [63]:
a=400
a//=13
print(a)

30


* Comparision Operator :- ( == , != , >= , <= , < , > )
> In Python Comparison of Relational operators compares the values. It either returns True or False according to the condition.

In [86]:
print(10==10) # Equal to: True if both operands are equal
print(10==20) # return false
print(10!=20) # Not equal to – True if operands are not equal	
print(10!=10) # return false
print(40>30)  # True if the left operand is greater than the right
print(30>50)  # return false
print(10<30)  # True if the left operand is less than the right
print(50<20)  # return false
print(10>=10) # True if the left operand is greater than or equal to the right
print(10>=20) # return false
print(20<=20) # True if the left operand is less than or equal to the right
print(20<=15) # false

True
False
True
False
True
False
True
False
True
False
True
False


* Logical Operator :- ( AND , OR , NOT )
> Python Logical operators perform Logical AND, Logical OR, and Logical NOT operations. It is used to combine conditional statements.

* Logical AND
1. True  - True  = True
2. True  - False = False
3. False - True  = False
4. False - False = False

In [90]:
print(10==10 and 20!=10)
print(10!=20 and 10>30)
print(20<50 and 20>10)
print(20<10 and 10!=10)

True
False
True
False


* Logical OR
1. True  - True  = True
2. True  - False = True
3. False - True  = True
4. False - False = False

In [91]:
print(10!=30 or 10==10)
print(50>20 or 20<10)
print(20!=20 or 20==20)
print(50>60 or 10!=10)

True
True
True
False


* Logical NOT 
1. True - False
2. False - True

In [94]:
print(not(20==20))
print(not(20!=20))

False
True


* Indentity Operator :- ( is , is not )
> The Identity operator are used to compare the memory addresses of two objects. Python has two identity operator.
1. 'is' 
> The 'is' operator returns 'true' if two varibles point to the same object in memory and otherwise, 'false'.
2. 'is not' 
> The 'is not' operator returns 'true' if two variables don't point to the same object in memory and otherwise, 'false'.

In [1]:
print('python' is 'python')

True


<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
C:\Users\hp\AppData\Local\Temp\ipykernel_4832\4152110594.py:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
  print('python' is 'python')


In [2]:
print('python' is 'Python')

False


<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
C:\Users\hp\AppData\Local\Temp\ipykernel_4832\1606315776.py:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
  print('python' is 'Python')


In [5]:
print(10 is not 12)

True


<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
C:\Users\hp\AppData\Local\Temp\ipykernel_4832\1994554126.py:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  print(10 is not 12)


In [6]:
print(20 is not 20)

False


<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
C:\Users\hp\AppData\Local\Temp\ipykernel_4832\166696429.py:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  print(20 is not 20)


* This is warning: - 
<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
C:\Users\hp\AppData\Local\Temp\ipykernel_4832\166696429.py:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  print(20 is not 20)

In [8]:
print('python'=='python')
print('python'=='Python')

True
False


In [11]:
print(20!=10)
print(10!=10)

True
False


* Membership Operator :- ( in , not in)
> Membership Operator are used to test if a sequence is presented in an object.
Python has two membership operator
1. in ( return 'true' if a sequence value is persent in the object)
2. not in ( return 'true' if a sequence value is not parsent in the object)

In [12]:
print('python' in 'python is a funny language.')

True


In [14]:
print('python' in 'Java is a funny language.')

False


In [15]:
print('python' not in 'javascript is client side lang.')

True


In [16]:
print('python' not in 'python is a funny lang.')

False


* Bitwise Operator :- ( ~ , | , & , >> , << )
> Bitwise Operator are used to compare (binary) numbers.

In [17]:
a=10
~a

-11

In [19]:
a=11
a|12

15

In [21]:
a=10
a>>2


2

In [22]:
a=10
a<<3


80